scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamWriter.scala

```diff
@@ -55,17 +56,18 @@ class BeamWriter(val beamMode: BeamMode) extends Writer {
     tempSources.get(initial) match {
       case Some(TempSource(path, coder)) =>
         val c: Coder[T] = coder.asInstanceOf[Coder[T]]
+
+        // Single dir by default just matches the dir, we need to match files inside
+        val matchedResources = FileSystems.`match`(s"$path*").metadata().asScala
+        val records = matchedResources.iterator.flatMap { resource =>
+          val is = Channels.newInputStream(FileSystems.open(resource.resourceId()))
+          InputStreamIterator.closingIterator(is, c)
+        }.toList
```

Comment:
this needs to be inside the Future or it will synchronously block other work.

```scala
import scala.concurrent.blocking

Future {
  blocking {
    // make the list here and the blocking construct will not block one of the CPU thread pools.

    ...
  }
}
```
see the discussion:

https://docs.scala-lang.org/overviews/core/futures.html